From 659748ea63a05fc76f3c552dfdc840d12ec25a11 Mon Sep 17 00:00:00 2001
From: Jian He <jianhe@apache.org>
Date: Fri, 8 May 2015 15:10:38 -0700
Subject: [PATCH 1322/2848] YARN-3604. Fixed ZKRMStateStore#removeApplication
 to also disable watch. Contributed Zhihai Xu
 (cherry picked from commit
 d0b73287c26983f32b0efe01862150e9fd28086f)

(cherry picked from commit 5a1c2fed349fd3e8a53300f574a952f0c3f622b6)

Change-Id: I274f41e999c80807e6030e117553e7d83513d453
---
 .../resourcemanager/recovery/ZKRMStateStore.java   |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/recovery/ZKRMStateStore.java b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/recovery/ZKRMStateStore.java
index 68c28df..b25d3b4 100644
--- a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/recovery/ZKRMStateStore.java
+++ b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/recovery/ZKRMStateStore.java
@@ -834,8 +834,8 @@ public synchronized void deleteStore() throws Exception {
   public synchronized void removeApplication(ApplicationId removeAppId)
       throws Exception {
     String appIdRemovePath = getNodePath(rmAppRoot, removeAppId.toString());
-    if (existsWithRetries(appIdRemovePath, true) != null) {
-      deleteWithRetries(appIdRemovePath, true);
+    if (existsWithRetries(appIdRemovePath, false) != null) {
+      deleteWithRetries(appIdRemovePath, false);
     }
   }
 
-- 
1.7.9.5

